src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb

```diff
@@ -527,7 +527,7 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.NavigationBar
                 newDocument.Project.Solution.Workspace.ApplyDocumentChanges(newDocument, cancellationToken)
 
                 ' WARNING: now that we've edited, nothing can check the cancellation token from here
```

Comment:
Comment is stale, and perhaps worrisome. The concern here is this is doing a single transaction of two steps: editing the file we're in and then also navigating to what we just generated. Historically the editor had some issues with rolling back transactions (see HACK_TextUndoTransactionThatRollsBackProperly.cs in our codebase) so this might have been essentially working around that: once we started editing files we were past a point of no-return.

If you dig into old source control and this was written before that HACK_TextUndoTransactionThatRollsBackProperly.cs then I'd say it's _probably_ safe to pass the cancellation token and delete the WARNING line -- the warning is no longer applicable since we fixed the underlying issue. If it was written in the other order historically...dunno. Maybe stick a big delay into the call below so you can trigger a cancellation yourself and see what the failure mode looks like? If it seems to cancel safely then I'd say delete the comment and ship it.

Reply:
ok.  i observed nothing wrong cancelling this.  will remove comment :)